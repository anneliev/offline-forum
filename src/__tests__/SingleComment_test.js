import React from 'react';
import { mount } from 'enzyme';
import SingleComment from '../components/SingleComment';

describe('single comment tests', () => {
  const aComment = 
    {
      comment: "A comment from Zac",
      id: "1",
      postId: 1,
      author: "Zac",
      date: "2018-01-29",
      currentPersona: "Zac",
      
    };

  it('should show a single comment', () => {
    const clickFn = jest.fn();
    const wrapper = mount(<SingleComment {...aComment} onClick={clickFn} />);
    expect(wrapper.find('p').first().text()).toContain("A comment from Zac");
  });

  it('should have id for onClick', () => {
    const clickFn = jest.fn();
    const wrapper = mount(<SingleComment {...aComment} onClick={clickFn} />);
    wrapper.find('button').simulate('click');
    expect(clickFn).toHaveBeenCalledWith('1');
  });

  it('should match snapshot', () => {
    const clickFn = jest.fn();
    const wrapper = mount(<SingleComment {...aComment} onClick={clickFn} /> );
    expect(wrapper.html()).toMatchSnapshot();
  })

});

