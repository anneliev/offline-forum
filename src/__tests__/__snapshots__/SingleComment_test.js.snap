// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single comment tests should match snapshot 1`] = `"<div class=\\"py-2 border-b relative\\"><button class=\\"bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full absolute pin-t pin-r my-3\\" data-test=\\"button\\">x</button><p class=\\"text-grey-darker mb-4\\">A comment from Zac</p><p class=\\"text-grey-dark mb-4\\">Posted by: Zac @ 2018-01-29</p></div>"`;
